### PR TITLE
life service implementation

### LINKED ISSUE #2: Use Case: Heart Rate Sensor
Sensor should be able to run on the background
Each HR values with timestamp should be stored in a local database

### LINKED ISSUE #12: Throws Error when try record hr values
When try record HR Value from it throw `Connection to service is not available for package 'com.harman.wearosapp' and action 'hs.versionclient.BIND'.`
I'm using Android Studio Beta becouse it suport HR censor value mocking so might be issue with studio itself

### PR DESCRIPTION
**Pull request by EntyR**

### CODE REVIEW

File: app/src/main/java/com/harman/wearosapp/app/hr_service/HealthServicesManager.kt [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+import android.content.ComponentName
+import android.content.Context
+import android.util.Log
+import androidx.concurrent.futures.await
+import androidx.health.services.client.HealthServicesClient
+import androidx.health.services.client.data.DataType
+import androidx.health.services.client.data.PassiveMonitoringConfig
+
+class HealthServicesManager(
```
Comment (AlexeyKolosov) [inline app/src/main/java/com/harman/wearosapp/app/hr_service/HealthServicesManager.kt]: I think it is better to move this to data layer

File: app/src/main/java/com/harman/wearosapp/app/hr_service/HRBroadcastReceiver.kt [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+import android.util.Log
+import androidx.health.services.client.data.DataType
+import androidx.health.services.client.data.PassiveMonitoringUpdate
+import com.harman.wearosapp.domain.use_cases.HRUseCase
+import kotlinx.coroutines.GlobalScope
+import kotlinx.coroutines.launch
+import org.koin.core.component.KoinComponent
+import org.koin.core.component.get
+
+class HRBroadcastReceiver() : BroadcastReceiver(
```
Comment (AlexeyKolosov) [inline app/src/main/java/com/harman/wearosapp/app/hr_service/HRBroadcastReceiver.kt]: I think it is better to move this to data layer

File: app/src/main/AndroidManifest.xml [hunk 3]
```diff
@@ -13,7 +21,17 @@
         android:icon="@mipmap/ic_launcher"
         android:label="@string/app_name"
         android:supportsRtl="true"
-        android:theme="@android:style/Theme.DeviceDefault">
+        android:name=".HeartRateApplication"
+        android:theme="@style/AppTheme">
+        <receiver
```
Comment (AlexeyKolosov) [inline app/src/main/AndroidManifest.xml]: After moving HR manager and receiver to data layer please move this too into manifest for the data layer

File: app/src/main/java/com/harman/wearosapp/app/hr_service/HRBroadcastReceiver.kt [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    override fun onReceive(context: Context, intent: Intent) {
+        Log.d(TAG, "onReceive: Received")
+        val lastDataPoint = (PassiveMonitoringUpdate.fromIntent(intent) ?: return)
+            .dataPoints
+            .filter { it.dataType == DataType.HEART_RATE_BPM }
+            .filter {
+                it.value.asDouble() > 0
+            }
+            .maxByOrNull { it.endDurationFromBoot } ?: return
+        GlobalScope.launch {
```
Comment (AlexeyKolosov) [inline app/src/main/java/com/harman/wearosapp/app/hr_service/HRBroadcastReceiver.kt]: please specify a background dispatcher here

File: app/src/main/java/com/harman/wearosapp/app/hr_service/HealthServicesManager.kt (hunk 1 above)
Comment (EntyR) [reply app/src/main/java/com/harman/wearosapp/app/hr_service/HealthServicesManager.kt]: done

File: app/src/main/java/com/harman/wearosapp/app/hr_service/HRBroadcastReceiver.kt (hunk 2 above)
Comment (EntyR) [reply app/src/main/java/com/harman/wearosapp/app/hr_service/HRBroadcastReceiver.kt]: done

File: app/src/main/AndroidManifest.xml (hunk 3 above)
Comment (EntyR) [reply app/src/main/AndroidManifest.xml]: done

File: app/src/main/java/com/harman/wearosapp/app/hr_service/HRBroadcastReceiver.kt (hunk 4 above)
Comment (EntyR) [reply app/src/main/java/com/harman/wearosapp/app/hr_service/HRBroadcastReceiver.kt]: done

**Review by AlexeyKolosov — APPROVED** [review]

**Pull request merged**